gnocchi/opts.py

```diff
@@ -190,6 +190,43 @@ def list_opts():
                 default=10,
                 help='Delay between flushes'),
         )),
+        ("amqp1d", (
+            cfg.HostAddressOpt('host',
+                               default='localhost',
+                               required=True,
+                               help='The listener IP for amqp1.0 '),
+            cfg.PortOpt('port',
+                        default=5672,
+                        required=True,
+                        help='The port for amqp 1.0'),
+            cfg.StrOpt(
+                'topic',
+                default='u/collectd/telemetry',
+                required=True,
+                help='Topic name to listen to for  1.0'),
+            cfg.StrOpt(
+                'data_source',
+                default='collectd',
+                required=True,
+                help='Data source for amqp1d(currently supporting '
+                     ' only collectd)'),
+            cfg.StrOpt(
+                'resource_name',
```

Comment:
`resource_type` then? It's not a name but a type IIUC

Reply:
I was trying to have resource_name( resource_type name)  in config, so the code will generate resource_type by name. Is that approach not correct?

Reply:
It is, is just don't think the name of the config option `resource_name` is explicit. `resource_type` or `resource_type_name` sounds better to me as it makes it clear it's the name of the resource type that I am defining.